libyul/YulString.h

```diff
@@ -77,11 +84,17 @@ class YulStringRepository: boost::noncopyable
 			hash ^= c;
 		}
 
+		unsigned char * byte = new unsigned char[8];
```

Comment:
This is evil and leaks memory ;-) (you may want to have a look at, e.g., https://stackoverflow.com/questions/6500313/why-should-c-programmers-minimize-use-of-new - you should hardly ever in fact use ``new``).

I think the easiest way to do what you want here would be to just *not* declare the ``s`` argument to this function as ``const`` and use:

```
for(int i = 0; i < 8; i++)
{
    hash *= 1099511628211u;
    hash ^= s & 0xFF;
    s >>= 8;
}
```
Or you could even "cheat" and just do ``hash *= 1099511628211u; hash ^= s;`` - that probably won't hurt too much either (although it's not "proper" hashing anymore that way).